src/trace.rs

```diff
@@ -370,27 +373,37 @@ impl TraceContext {
     }
 
     pub fn new_openssl_agent(&mut self, descriptor: &AgentDescriptor) -> Result<AgentName, Error> {
-        let agent_name = self.add_agent(Agent::new_openssl(descriptor, self.claimer.clone())?);
+        // [TODO] There might be a cleaner way of doing this, e.g., if we could store a type
+        // instantiation in an enum, but this works
+        let agent = match descriptor.put_type {
+            PUTType::OpenSSL => Agent::new::<OpenSSL>(descriptor, self.claimer.clone())?,
+            PUTType::WolfSSL => Agent::new::<WolfSSL>(descriptor, self.claimer.clone())?,
+        };
```

Comment:
I think we need at some point a cleaner method, else we can not disable WolfSSL/OpenSSL via a feature flag.

Reply:
Hmmm, that's a good point. I wasn't in my list of requirements but it should have been :(